alea/submitters/htcondor.py

```diff
@@ -123,41 +127,29 @@ def requirements(self):
 
         return _requirements
 
-    def _validate_template_path(self):
-        """Validate the template path."""
-        if self.template_path is None:
-            raise ValueError("Please provide a template path.")
-        # This path must exists locally, and it will be used to stage the input files
-        if not os.path.exists(self.template_path):
-            raise ValueError(f"Path {self.template_path} does not exist.")
-
-        # Printout the template path file structure
-        logger.info("Template path file structure:")
-        for dirpath, dirnames, filenames in os.walk(self.template_path):
-            for filename in filenames:
-                logger.info(f"File: {filename} in {dirpath}")
-        if self._contains_subdirectories(self.template_path):
-            logger.warning(
-                "The template path contains subdirectories. All templates files will be tarred."
-            )
-
     def _tar_h5_files(self, directory, template_tarball="templates.tar.gz"):
         """Tar all .h5 templates in the directory and its subdirectories into a tarball."""
         # Create a tar.gz archive
         with tarfile.open(template_tarball, "w:gz") as tar:
-            # Walk through the directory
-            for dirpath, dirnames, filenames in os.walk(directory):
-                for filename in filenames:
-                    if filename.endswith(".h5"):
-                        # Get the full path to the file
-                        filepath = os.path.join(dirpath, filename)
-                        # Add the file to the tar
-                        # Specify the arcname to store relative path within the tar
-                        tar.add(filepath, arcname=os.path.basename(filename))
+            tar.add(directory, arcname=os.path.basename(directory))
```

Comment:
This could in principle add things that are not needed (i.e. everything not ending on .h5, like pdf_cache or something), right? I also have a faint memory of Tim choosing the flat folder structure because otherwise it wouldn't work on the computing node but I might be wronog -- have you checked though?

Reply:
If you run into the code, you will find that only templates are copied to `self.templates_tarball_dir`.